### PR TITLE
Fixes for CuDNNDropoutOp

### PR DESCRIPTION
**Pull request by lukeyeager**

### CODE REVIEW

File: caffe2/operators/dropout_op_cudnn.cc [hunk 1]
```diff
@@ -114,10 +115,10 @@ bool CuDNNDropoutOp::DoRunWithType() {
         1));
     // get the reserve space we need
     CUDNN_ENFORCE(cudnnDropoutGetReserveSpaceSize(
-        data_desc_, reinterpret_cast<size_t*>(&reserve_space_size_in_bytes_)));
```
Comment (lukeyeager) [inline caffe2/operators/dropout_op_cudnn.cc:117]: This was bad

File: caffe2/operators/dropout_op_cudnn.cc [hunk 2]
```diff
@@ -5,6 +5,33 @@
...
+
+// Round up N to nearest multiple of A.
+size_t roundUp(size_t n, size_t a) {
+  return n + ((-n % a) + a) % a;
+}
+
+constexpr size_t alignmentBytes = 128;
+
+// Returns the index into mask_and_states where states_data begins.
+// Ensures that states_data is properly aligned.
```
Comment (lukeyeager) [inline caffe2/operators/dropout_op_cudnn.cc:18]: This is helpful

**Review by slayton58 — APPROVED** [review]
Fixes a nasty crash bug, and makes the code much easier to follow.

**Comment by akyrola** [comment]
So the issue was int/size_t?

**Pull request closed without merging**